datafusion/core/tests/physical_optimizer/filter_pushdown/mod.rs

```diff
@@ -433,6 +433,117 @@ async fn test_topk_dynamic_filter_pushdown() {
     );
 }
 
+#[tokio::test]
+async fn test_hashjoin_dynamic_filter_pushdown() {
```

Comment:
Can we add some tests for multiple joins? Such as

```
Join (t1.a = t2.b)
/        \
t1    Join(t2.c = t3.d)
        /    \
       t3   t2
```

Reply:
Such test can check 
1. dynamic filters are pushed down to right scan node 
2. dynamic filters aren't missed during pushdown